tests/functional/r/redefined_except_handler.py

```diff
@@ -0,0 +1,69 @@
+"""https://github.com/PyCQA/pylint/issues/5370"""
+
+try:
+    pass
+except ImportError as err:
+    try:
+        pass
+    except ImportError as err:  # [redefined-outer-name]
+        pass
+    print(err)
+
+try:
+    pass
+except ImportError as err:
+    try:
+        pass
+    except ImportError as err2:
+        pass
+    print(err)
+
+try:
+    try:
+        pass
+    except ImportError as err:
+        pass
+except ImportError:
+    try:
+        pass
+    except ImportError as err:
+        pass
+    print(err)
+
+
+try:
+    try:
+        pass
+    except ImportError as err:
+        pass
+except ImportError as err:
+    try:
+        pass
+    except ImportError:
+        pass
+    print(err)
+
+try:
+    pass
+except ImportError as err:
+    try:
+        pass
+    except ImportError as err2:
+        try:
+            pass
+        except ImportError as err:  # [redefined-outer-name]
+            pass
+    print(err)
+
+
+class CustomException(Exception):
+    """https://github.com/PyCQA/pylint/issues/4434"""
+
+
+def func():
+    """Override CustomException by except .. as .."""
+    try:
+        raise CustomException('Test')  # [used-before-assignment]
```

Comment:
Why do we have a ``used-before-assignment`` here ? ``CustomException`` is defined line 59 as far as I understand ? Are you underling a false positive ?

Reply:
nvm, I just read https://github.com/PyCQA/pylint/issues/4434#issuecomment-1003838019. I think python is counterintuitive here 😄